src/pages/ReimbursementAccount/ACHContractStep.js

```diff
@@ -36,74 +38,105 @@ const propTypes = {
+                }
+
+                if (!ValidationUtils.isRequiredFulfilled(values[`beneficialOwner.${ownerKey}.state`])) {
+                    errors[`beneficialOwner.${ownerKey}.state`] = this.props.translate('bankAccount.error.addressState');
+                }
+
+                if (!ValidationUtils.isRequiredFulfilled(values[`beneficialOwner.${ownerKey}.zipCode`])
+                    || !ValidationUtils.isValidZipCode(values[`beneficialOwner.${ownerKey}.zipCode`])) {
+                    errors[`beneficialOwner.${ownerKey}.zipCode`] = this.props.translate('bankAccount.error.zipCode');
+                }
+            });
         }
 
-        const errors = {};
-        _.each(this.requiredFields, (inputKey) => {
-            if (ValidationUtils.isRequiredFulfilled(this.state[inputKey])) {
-                return;
-            }
+        if (!ValidationUtils.isRequiredFulfilled(values.acceptTermsAndConditions)) {
+            errors.acceptTermsAndConditions = this.props.translate('common.error.acceptedTerms');
+        }
 
-            errors[inputKey] = true;
-        });
-        BankAccounts.setBankAccountFormValidationErrors({...errors, beneficialOwnersErrors});
-        return _.every(beneficialOwnersErrors, _.isEmpty) && _.isEmpty(errors);
+        if (!ValidationUtils.isRequiredFulfilled(values.certifyTrueInformation)) {
+            errors.certifyTrueInformation = this.props.translate('beneficialOwnersStep.error.certify');
+        }
+
+        return errors;
     }
 
-    removeBeneficialOwner(beneficialOwner) {
+    /**
+     * @param {int} ownerKey - ID connected to the beneficial owner identity form
+     */
+    removeBeneficialOwner(ownerKey) {
         this.setState((prevState) => {
-            const beneficialOwners = _.without(prevState.beneficialOwners, beneficialOwner);
+            const beneficialOwners = _.without(prevState.beneficialOwners, ownerKey);
 
-            // We set 'beneficialOwners' to null first because we don't have a way yet to replace a specific property without merging it.
-            // We don't use the debounced function because we want to make both function calls.
-            BankAccounts.updateReimbursementAccountDraft({beneficialOwners: null});
-            BankAccounts.updateReimbursementAccountDraft({beneficialOwners});
+            FormActions.setDraftValues(ONYXKEYS.REIMBURSEMENT_ACCOUNT, {beneficialOwners});
 
-            // Clear errors
-            BankAccounts.setBankAccountFormValidationErrors({});
             return {beneficialOwners};
         });
     }
 
     addBeneficialOwner() {
-        this.setState(prevState => ({beneficialOwners: [...prevState.beneficialOwners, {}]}));
+        this.setState((prevState) => {
+            // Each beneficial owner is assigned a unique key to connect to the correct Identity Form.
+            // This is because each Identity form is dynamically rendered by retrieving stored form values for each ownerKey.
```

Comment:
This reads as circular logic to me.

_We assign each owner a key because each form needs it's own key._ 

Maybe better to explain why each form needs it's own key at all? I think that's what I'm missing here.